### PR TITLE
Rspe controllers

### PR DESCRIPTION
**Pull request by shella12**
- [ ] Created a Request spec file for all controllers.

- [ ] Made sure that for each action following checks are performed:

- [ ] If response status was correct.

- [ ] If a correct template was rendered.

- [ ] If the response body includes correct placeholder text.

### CODE REVIEW

File: spec/requests/posts_spec.rb [hunk 1]
```diff
@@ -3,7 +3,21 @@
...
+      expect(response).to have_http_status(:success)
+    end
+  end
+
+  describe 'GET users/:id/post' do
+    it 'returns http success' do
+      get '/users/1/posts'
       expect(response).to have_http_status(:success)
     end
   end
```
Comment (julie-ify) [inline spec/requests/posts_spec.rb]: - [ ] Kindly organize your tests here. You should only have two `describe` groupings for post index and show pages.

  ```
  describe 'GET #index' do
    it '...' do
      ...
    end
    ....
  end

  describe 'GET #show' do
    it '...' do
      ...
    end
    ...
  end
  ```

- [ ] Besides checking for the status you should also check for text on the respective pages and ensure the correct template is being rendered:

  ```
  expect(response).to render_template(:show)
  ```
  ```
  expect(response.body).to include("Welcome to user index page")
  ```

File: spec/requests/user_spec.rb [hunk 2]
```diff
@@ -7,4 +7,11 @@
       expect(response).to have_http_status(:success)
     end
   end
+
+  describe 'GET /user/:id' do
+    it 'returns http success' do
+      get '/users/1'
+      expect(response).to have_http_status(:success)
+    end
+  end
```
Comment (julie-ify) [inline spec/requests/user_spec.rb]: - [x] Kindly add more tests in to check for templates and text from the pages.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @shella12,

Good job so far!

## Highlights!
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/requests/user_spec.rb (hunk 2 above)
Comment (shella12) [reply spec/requests/user_spec.rb]: - [ ] Those tests have already been created in a separate folder `view`

File: spec/requests/posts_spec.rb (hunk 1 above)
Comment (shella12) [reply spec/requests/posts_spec.rb]: - [ ] Tests for all the html.erb files are already been created and put in a folder `view`
- [ ] For checking the routes I have put already put tests for checking show and index routes

**Review by Nemwel-Boniface — COMMENTED** [review]
# Hello @shella12   👋,

I am Nemwel your reviewer.


## _This is an additional review as a result of the second opinion request._

I have inspected your project and checked the review that was performed by @julie-ify and I found that it was in line with the [project requirements](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/controller_specs_project.md) for this project (Blog app - controllers specs).

In your spec/requests folder, all the tests that are currently written only check for the response status and none of them check for:
1. If a correct template was rendered.
2. If the response body includes correct placeholder text.

As these two tests are not written in both your "posts_spec.rb" and "user_spec.rb", kindly ensure that you write the tests and add them in your spec files just as @julie-ify  requested.

In case anything is not 100% clear tag me @Nemwel-Boniface so that I get the notification.

Keep rocking 🚀!

**Comment by shella12** [comment]
@Nemwel-Boniface 

as I have already mentioned does tests are present in the specs folder under the view folder, I have created two separate folders for controller and view tests and those two folders are `requests` and `views` respectively. Here is a screenshot

Screenshot 2023-03-17 at 4 40 57 PM

**Review by Nemwel-Boniface — COMMENTED** [review]
Hello @shella12 

We need to make sure that we stick to the project requirements. You are required to put all the tests in the requests folder as per the project requirements.

image

Here is a situation where a client gives you his/ her requirements to you as the developer for their app. You then decide to go against the project requirements that were given by the client, which definitely will not sit right with him/ her. I hope this analogy makes sense in this scenario?

**Kindly note:** Checking through the codebase at the time the reviewer reviewed the project, this is what is there in the post_spec.rb

image

Likewise, this is what is present in the user_spec.rb

image

I hope it is now clear to you?

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED 🍾✔🎉**
Hello @shella12 👋
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- You documented your work professionally.✔
- You asked for the changes in a polite and friendly way.
- Your linters are passing all the checks.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**